python/ray/data/dataset.py

```diff
@@ -2463,6 +2463,124 @@ def _validate_test_size_int(self, test_size: int, ds: "Dataset") -> int:
             )
         return ds_length
 
+    @PublicAPI(stability="alpha", pi_group=SMJ_API_GROUP)
+    def streaming_train_test_split(
+        self,
+        test_proportion: float,
+        *,
+        split_type: Literal["hash", "bernoulli"] = "bernoulli",
+        key_column: Optional[str] = None,
+        seed: Optional[int] = None,
+        **ray_remote_kwargs,
+    ) -> Tuple["Dataset", "Dataset"]:
+        """split the dataset into train and test subsets in a streaming manner.
+        This method is recommended for large datasets.
+
+        The split type can be either "hash" or "bernoulli".
+        - "hash": The dataset is split into train and test subsets based on the hash of the key column.
+        - "bernoulli": The dataset is split into train and test subsets based on the Bernoulli distribution.
+
+        Examples with Bernoulli split:
+
+            >>> import ray
+            >>> ds = ray.data.range(8, override_num_blocks=1)
```

Comment:
I had to override the num_blocks when the data is very small, otherwise it will create one block per record and the `bernulli` split will put all records into the same bucket